### PR TITLE
Fixes Missing Map Icons

### LINKED ISSUE #435: Location Pins Missing in Google Maps API in Encounter page
**Expected behavior**
`/encounters/encounter.jsp` shows a map with a pin that corresponds to the Encounter's Location ID or coordinates.

**Current behavior**
`/encounters/encounter.jsp` shows a map but without a pin even when the Encounter's Location ID or coordinates are present. 

**Investigation notes** 
Example encounter: https://www.whiskerbook.org/encounters/encounter.jsp?number=a05238ea-db03-497f-93a4-a16d4dc26dfd

The pin is missing from `/encounters/encounter.jsp` when using Chrome or Safari, but is present in `/submit.jsp`. Brave and Firefox still display the pin. 

Re-check in one week to see if Brave stops displaying the pin on the Encounter page. Possible Google Maps API issue.

From @holmbergius: This week I have seen progressively more and more Google Maps pins in Wildbook disappear, and I believe it is a Google Maps library-related upgrade:
https://developers.google.com/maps/documentation/javascript/advanced-markers/migration#:~:text=As%20of%20February%2021st%2C%202024,Marker%20is%20deprecated.

**Community link**
https://community.wildme.org/t/no-more-pin-in-maps/2901

### PR DESCRIPTION
**Pull request by holmbergius**
Wildbook color-coded its Google Maps pins using a Google Charts API that was turned off in 2019 yet still kept working until...now-ish. With that API now completely turned off for good, our maps pins are missing.

This is the old call:
icon: 'https://chart.googleapis.com/chart?chst=d_map_pin_letter&chld=<%=markerText%>|<%=colorToUseForMarker%>',
 
The short-term fix is to replace the icon attribute with a Materials font family character label. This actually adds a bit of diversity in the symbols we can use and ended up being somewhat of an improvement in that we can tweak more of the infill, color, size, and display of the icon. 

Longer-term, we need to migrate Marker to AdvancedMarkerElement, but that is a major overhaul.

Tested successfully in Chrome and Firefox.

PR fixes #435 
image

image

### CODE REVIEW

**Comment by holmbergius** [comment]
@naknomum Would love your thoughts on the fix.

File: src/main/webapp/encounters/encounter.jsp [hunk 1]
```diff
@@ -1293,9 +1294,17 @@ if(CommonConfiguration.showProperty("showCountry",context)){
            	%>
 
        marker = new google.maps.Marker({
-    	   icon: 'https://chart.googleapis.com/chart?chst=d_map_pin_letter&chld=<%=markerText%>|<%=haploColor%>',
+    	   //icon: 'https://chart.googleapis.com/chart?chst=d_map_pin_letter&chld=<%=markerText%>|<%=haploColor%>',
+    	   icon: " ",
```
Comment (naknomum) [inline src/main/webapp/encounters/encounter.jsp:1298]: am guessing the space must be needed here (versus just dropping the `icon` element from the json entirely).

**Review by naknomum — APPROVED** [review]
seems like a solid workaround until we update to advanced marker api.

File: src/main/webapp/encounters/encounter.jsp (hunk 1 above)
Comment (holmbergius) [reply src/main/webapp/encounters/encounter.jsp:1298]: I believe the space is needed according to some reading I did on this issue. But I also didn't test without it.

**Pull request merged**